tools/registry.py

```diff
@@ -434,6 +434,16 @@ def add(self, module, override=False):
     metadata["versions"].sort(key=Version)
     json_dump(metadata_path, metadata)
 
+  def update_integrity(self, module_name, version):
+    """Update the SRI hashes of the source.json file of module at version."""
+    source = self.get_source(module_name, version)
+    source_path = self.get_source_path(module_name, version)
+    source["integrity"] = integrity(download(source["url"]))
+    patch_base = source_path.parent / "patches"
+    for patch_name in source.get("patches", []):
+        source["patches"][patch_name] = integrity(read(patch_base / patch_name))
```

Comment:
How about using the `patches` directory as the source of truth for regenerating the `patches` section? That would help if users added patches to an existing module.

Reply:
Sounds good, these edge cases look dubious anyway.